### ISSUE TITLE
false positive: semicolon after a templated lambda

### ISSUE THREAD

**Issue by dvirtz** [cpplint/cpplint#385]
Running cpplint on 
```
int main() {
  auto identity = []<typename T>(T&& t) {
    return t;
  };
}
```
produces
```
lambda.cpp:4:  You don't need a ; after a }  [readability/braces] [Error]
```
